File: C++/345. Reverse Vowels of a String.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int i=0,j=s.size()-1;
        unordered_set<char>a={'a','e','i','o','u','A','E','I','O','U'};
        while(i<j){
            if(a.count(s[i])==0)
            i++;
            if(a.count(s[j])==0)
            j--;
            if(a.count(s[i])!=0 && a.count(s[j])!=0){
                swap(s[i],s[j]);
                i++;
                j--;

            }
        }
        return s;
    }
};

File: JAVA/345. Reverse Vowels of a String.java
class Solution {
    public String reverseVowels(String s) {
        char[] a = s.toCharArray();
        char c;
        int i=0,j=a.length-1;
        while(i<j){
            if(is(a[i])==false)
            i++;
            if(is(a[j])==false)
            j--;
            if(is(a[i])==true&&is(a[j])==true){
                c=a[i];
                a[i++]=a[j];
                a[j--]=c;
            }
        }
        return new String(a);
    }
    private boolean is(char c){
        return(c=='a'||c=='e'||c=='i'||c=='o'||c=='u'||c=='A'||c=='E'||c=='I'||c=='O'||c=='U');
    }
}